utils/misc.ml

```diff
@@ -564,11 +579,20 @@ module Stdlib = struct
       let rec all_unit = function
         | [] -> return ()
         | t :: ts -> t >>= fun () -> all_unit ts
+
+      module Syntax = struct
+        let[@inline] (let+) t f = map f t
+        let[@inline] (and+) a b = a >>= fun a -> b >>= fun b -> return (a,b)
```

Comment:
I think it's worth defining a `both` outside the `Syntax` module and implementing `and+` in terms of it.